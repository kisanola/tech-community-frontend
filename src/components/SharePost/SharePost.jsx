import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FacebookShareButton, TwitterShareButton } from 'react-share';
import PropTypes from 'prop-types';
import Modal from '../Modal/Modal';
import './SharePost.scss';
import facebookLogo from '../../assets/images/facebook_logo.png';
import twitterLogo from '../../assets/images/twitter_logo.png';

import { FRONTEND_BASE_URL } from '../../constants';

import shareAction from '../../actions/sharePostAction';


export class SharePost extends Component {
  handleClick = (postSlug, platform) => {
    const { share, handleClose } = this.props;
    share({ postSlug, platform });
    handleClose();
  };

  render() {
    const { show, handleClose, postSlug } = this.props;
    const POST_URL = `${FRONTEND_BASE_URL}/post/${postSlug}`;

    return (
      <Modal show={show} handleClose={handleClose}>
        <div className="share-post">
          <div
            role="presentation"
            className="social_share"
            id="twitter"
            onClick={() => this.handleClick(postSlug, 'twitter')}
          >
            <TwitterShareButton url={POST_URL}>
              <img
                src={twitterLogo}
                className="social_logo"
                alt="twitter_logo"
              />
              <p className="social_name">SHARE ON TWITTER</p>
            </TwitterShareButton>
          </div>

          <div
            role="presentation"
            className="social_share"
            id="facebook"
            onClick={() => this.handleClick(postSlug, 'facebook')}
          >
            <FacebookShareButton url={POST_URL}>
              <img
                src={facebookLogo}
                className="social_logo"
                alt="facebook_logo"
              />
              <p className="social_name">SHARE ON FACEBOOK</p>
            </FacebookShareButton>
          </div>
        </div>
      </Modal>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  share: (data) => dispatch(shareAction(data)),
});

SharePost.propTypes = {
  show: PropTypes.bool,
  postSlug: PropTypes.string,
  handleClose: PropTypes.func.isRequired,
  share: PropTypes.func,
};

SharePost.defaultProps = {
  show: false,
  postSlug: '',
  share: () => '',
};

export default connect(
  null,
  mapDispatchToProps,
)(SharePost);
