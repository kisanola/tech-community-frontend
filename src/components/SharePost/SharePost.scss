@import '../../styles/colors';

.share-post {
  background: #fff;
  border-radius: 3px;
  width: 50%;
  min-width: 300px;
  max-width: 600px;

  &__header {
    font-size: 1.5rem;
    color: $secondary;
    padding: 10px;
    text-align: center;
  }

  .SocialMediaShareButton {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .social_share {
    text-align: center;
    margin: 10px;
    padding: 3px 0;
    display: flex;
    justify-content: center;
    align-items: center;
    &:hover {
      cursor: pointer;
    }

    .social_logo {
      display: inline-block;
      height: 25px;
      width: 25px;
      margin: 0;
      padding: 2px;
      box-sizing: border-box;
    }
    .social_name {
      display: inline-block;
      margin: 0px 5px 0px 0;
      padding: 5px;
      vertical-align: top;
      font-weight: bold;
      color: #fff;
      font-size: 0.9em;
      font-family: arial;
      box-sizing: border-box;
    }
  }
  #twitter {
    background: #1fa1f1;
  }
  #twitter img {
    background: #fff;
    height: 20px;
    width: 20px;
    padding: 2px;
    margin-right: 5px;
    border-radius: 50%;
  }
  #facebook {
    background: #4266b2;
  }
}
