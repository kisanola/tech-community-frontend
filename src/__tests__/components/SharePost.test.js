import React from 'react';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import SharePost, {
  SharePost as SharePostComponent,
} from '../../components/SharePost/SharePost';

describe(`SharePost`, () => {
  const props = {
    show: true,
    handleClose: jest.fn(),
    share: jest.fn(),
  };
  it('should render `SharePost`', () => {
    const mockStore = configureMockStore([thunk]);
    const store = mockStore({});
    const wrapper = shallow(
      <Provider store={store}>
        <SharePost {...props} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should close SharePost modal when handleClose is hit', () => {
    const wrapper = mount(<SharePostComponent {...props} />);
    wrapper.find('#twitter').simulate('click');
    expect(wrapper.props().handleClose).toHaveBeenCalled();
  });
});
