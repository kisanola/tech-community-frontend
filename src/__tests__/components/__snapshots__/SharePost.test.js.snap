// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharePost should render \`SharePost\` 1`] = `ShallowWrapper {}`;
